OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs

```diff
@@ -128,11 +128,16 @@ public Actor[] Reinforce(Player owner, string[] actorTypes, CPos[] entryPath, in
 				}
 			}
 
-			for (var i = 1; i < entryPath.Length; i++)
+			// We can't move planes to the last waypoint yet,
+			// as that causes them to teleport to ground level
+			for (var i = 1; i < entryPath.Length - 1; i++)
 				Move(transport, entryPath[i]);
 
+			var dropoffPoint = entryPath.Last();
 			if (actionFunc != null)
 			{
+				Move(transport, dropoffPoint);
```

Comment:
If `entryPath` has one element then this will queue a `Move` to it that we don't want (it is excluded in the old code by starting the loop at index 1).